### PR TITLE
use struct `Resources` for `UsedResources` in tests

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Simple change but more readable and descriptive. 

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -61,7 +61,8 @@ func New(client client.Client) *Cache {
 	}
 }
 
-type Resources map[corev1.ResourceName]map[string]int64
+type Resources map[corev1.ResourceName]FlavorResources
+type FlavorResources map[string]int64
```
Comment (ahg-g) [inline pkg/cache/cache.go:65]: We have ResourceFlavor API, this type name is going to be confusing. 

I suggest we keep it as is, and just merge the changes to `cache_test.go`

Comment (kerthcet) [reply pkg/cache/cache.go:65]: This is also what I'm hesitate about, I'll follow you advices then.

Comment (kerthcet) [reply pkg/cache/cache.go]: updated.

**Pull request merged**